serde/src/test/kotlin/com/amazon/ionhiveserde/objectinspectors/IonStructToMapObjectInspectorTest.kt

```diff
@@ -18,86 +18,482 @@ package com.amazon.ionhiveserde.objectinspectors
-        assertEquals(ION.newInt(3), subject.getMapValueElement(struct, ION.newSymbol("c")))
-        assertNull(subject.getMapValueElement(struct, ION.newSymbol("d")))
+        val floatIntStruct = structs[5]
+        val floatList = listOf<Float>(2f, 4f, 8f, 16f)
+        assertEquals(ION.newInt(1), floatIntSubject.getMapValueElement(floatIntStruct, ION.newSymbol(floatList[0].toString())))
+        assertEquals(ION.newInt(2), floatIntSubject.getMapValueElement(floatIntStruct, ION.newSymbol(floatList[1].toString())))
+        assertEquals(ION.newInt(3), floatIntSubject.getMapValueElement(floatIntStruct, ION.newSymbol(floatList[2].toString())))
+        assertNull(floatIntSubject.getMapValueElement(floatIntStruct, ION.newSymbol(floatList[3].toString())))
+
+        val doubleIntStruct = structs[6]
+        val doubleList = listOf<Double>(2.0, 4.0, 6.0, 8.0)
+        assertEquals(ION.newInt(1), doubleIntSubject.getMapValueElement(doubleIntStruct, ION.newSymbol(doubleList[0].toString())))
+        assertEquals(ION.newInt(2), doubleIntSubject.getMapValueElement(doubleIntStruct, ION.newSymbol(doubleList[1].toString())))
+        assertEquals(ION.newInt(3), doubleIntSubject.getMapValueElement(doubleIntStruct, ION.newSymbol(doubleList[2].toString())))
+        assertNull(doubleIntSubject.getMapValueElement(doubleIntStruct, ION.newSymbol(doubleList[3].toString())))
+
+        val boolIntStruct = structs[7]
+        val boolList = listOf<Boolean>(true, false)
+        assertEquals(ION.newInt(1), booleanIntSubject.getMapValueElement(boolIntStruct, ION.newSymbol(boolList[0].toString())))
+        assertEquals(ION.newInt(2), booleanIntSubject.getMapValueElement(boolIntStruct, ION.newSymbol(boolList[1].toString())))
+
+        val decimalIntStruct = structs[8]
+        val decimalList = listOf<HiveDecimal>(HiveDecimal.create(1), HiveDecimal.create(2))
+        assertEquals(ION.newInt(1), decimalIntSubject.getMapValueElement(decimalIntStruct, ION.newSymbol(decimalList[0].toString())))
+        assertEquals(ION.newInt(2), decimalIntSubject.getMapValueElement(decimalIntStruct, ION.newSymbol(decimalList[1].toString())))
+
+        val dateIntStruct = structs[9]
+        val dateList = listOf<Date>(
+                Date.valueOf("1991-1-1"),
+                Date.valueOf("1992-2-3"))
+        assertEquals(ION.newInt(1), dateIntSubject.getMapValueElement(dateIntStruct, ION.newSymbol(dateList[0].toString())))
+        assertEquals(ION.newInt(2), dateIntSubject.getMapValueElement(dateIntStruct, ION.newSymbol(dateList[1].toString())))
+
+        val timestampIntStruct = structs[10]
+        val tsList = listOf<Timestamp>(
+                Timestamp.valueOf("2014-10-14 12:34:56.789"),
+                Timestamp.valueOf("2015-10-16 12:34:56.789"))
+        assertEquals(ION.newInt(1), timestampIntSubject.getMapValueElement(timestampIntStruct, ION.newSymbol(tsList[0].toString())))
```

Comment:
Assuming `java.sql.Timestamp.toString` provides the same strings you provided to `valueOf`, these aren't valid Ion Timestamps in text Ion. This is an example of where we need to be clear about the format of the data that is allowed in field names. An Ion reader won't parse these timestamps, and some other parser (like `java.sql.Timestamp.valueOf`) might not parse text Ion timestamps.

Reply:
Extracting a helper method here will help with integrating with Hive 3- Hive 3 no longer uses `java.sql` types for `Date` and `Timestamp` so any direct reference to those types has to be fixed on the `hive3` branch.

Reply:
@tgregg Good catch! With the new implementation, the timestamps ended up getting serialized with only day precision, so I have since updated the base struct to use serialized IonTimestamps, which then get properly deserialized as Timestamps.